p2p/src/peer_manager/peerdb/new_tried_addr_table/mod.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2021-2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::hash::{Hash, Hasher};
+
+use crypto::random::{make_pseudo_rng, Rng};
+use p2p_types::socket_address::SocketAddress;
+use serialization::{Decode, Encode};
+
+use self::table::Table;
+
+#[macro_use]
+mod table;
+
+// TODO: do we need the tables to be this big?
+const NEW_ADDR_TABLE_BUCKET_COUNT: usize = 1024;
+const TRIED_ADDR_TABLE_BUCKET_COUNT: usize = 256;
+const BUCKET_SIZE: usize = 64;
+
+/// The random key that is used to randomize bucket selection inside the tables.
+///
+/// It will be randomly generated once and stored in the peer db.
+#[derive(Hash, Encode, Decode, Copy, Clone)]
+pub struct RandomKey(u64);
+
+impl RandomKey {
+    pub fn new_random() -> Self {
+        RandomKey(make_pseudo_rng().gen::<u64>())
+    }
+}
+
+fn mix_random_key<T: Hash>(key: RandomKey, data: T) -> RandomKey {
+    RandomKey(calc_hash!(key.0, data))
+}
+
+pub struct NewTriedAddrTable {
```

Comment:
Please write a big comment explain what this is and how it works.

Reply:
Done. Let me know if it's enough.
Also, I renamed the struct to just `AddressTables`.

Reply:
It's OK now. But why is it thrown to the top of the file instead of it being on top of AddressTables?

Reply:
I've put the comment on top of AddressTables.